packages/yarnpkg-pnpify/sources/buildNodeModulesTree.ts

```diff
@@ -40,11 +42,18 @@ export type NodeModulesTree = Map<PortablePath, NodeModulesBaseNode | NodeModule
 
 export interface NodeModulesTreeOptions {
   pnpifyFs?: boolean;
+  nohoistPatterns?: Array<{relativeCwd: string, nohoistPatterns: Array<string>}>;
```

Comment:
I know that PnPify is supposed to be PnP implementation-agnostic, but I'd find it weird that `yarn pnpify tsc ...` and `YARN_PNP_MODE=loose yarn && yarn tsc ...` would use 2 different hoisting layouts when `nohoist` is used.

I think that our options to fix this would be:
- do nothing, just accept that things are going to be hoisted when running `yarn pnpify tsc`, even if `nohoist` is used.
- make a Yarn-specific integration inside the PnPify patch thing.
- make PnPify read and process the manifest to gather the `nohoist` patterns. I don't like this idea, as `nohoist` isn't a universal concept and other PnP implementors might want something different.

What do you think? 🤔

Reply:
To be fair I have just forgot about the pnpify :) Yeah, both of your concerns are revolving around how to pass `nohoist` settings from manifests to `buildNodeModulesTree` where they need to be prepared first and then passed to `hoist`. I yet to look one more time to this problem, this PR is a Draft for a reason, but you have well spotted one of the main problems that need to be solved

Reply:
The idea though is to make `nohoist` work for `pnpify`, NM Linker and loose mode PnP Linker, I don't want to leave surprises for the user. Of course it would be better to not pollute PnP with `nohoist`, but I'm yet to find a way how to do it, or whether it is possible at all, without too much hassle